watertap/unit_models/crystallizer.py

```diff
@@ -446,6 +473,14 @@ def eq_p_con2(b):
         def eq_p_con3(b):
             return b.properties_vapor[0].pressure == self.pressure_operating
 
+        @self.Constraint()
+        def eq_p_con4(b):
+            return b.properties_pure_water[0].pressure == self.pressure_operating
+
+        @self.Constraint()
+        def eq_p_con5(b):
+            return b.properties_pure_water[0].pressure_sat == self.pressure_operating
```

Comment:
I'm not sure I understand this. It looks like the pure water and the vapor streams are exactly the same, or am I missing something? If they are not, how are they different?

- The flow rates are the same
- The pressures are the same

Reply:
Yes thanks for asking as it's a bit confusing here and I'll describe it in the document. So you're right that the flow rates and pressure are the same, but the corresponding temperatures are not. 

The temperature of `properties_vapor` is in equilibrium with the solution from which it is generated, and the boiling point of the saturated brine is higher than that of pure water, leading to the superheated vapor going into the next effect (heat exchanger). The temperature of `properties_pure_water` is in equilibrium of pure water at the same pressure as depicted in `eq_p_con5`: `b.properties_pure_water[0].pressure_sat == self.pressure_operating`.

Hope it helps clarify!

Reply:
So you're bring the steam from the superheated state to its saturation point and condensing. So, in effect, what you have at the end is the saturated liquid, correct?

Reply:
Yes, I suppose this model naturally generates saturated liquid at the end, correspondingly related to the operating pressure/temperature, so that the salts can precipitate.